### DISCUSSION TITLE
Keyball 61 right and leftside not working together2

### DISCUSSION THREAD

**Discussion by klinsor** [Yowkees/keyball#509] [Q&A]
hello. I assembled keyball61 but i got one issue.... I hope you can help me. Coincidentally, it is the same issue as the person just before.

Everything else works normally, but only the trrs cable doesn't work properly.
When the trrs cable is not connected, it works well individually. However, when I connect the trrs cable, the keyboard on the other side does not work.
I don't know what the problem might be, so I took a picture of the Pro Micro and attached it, so I hope you can take a look to see if there is a problem.
![photo_2024-03-22_18-54-32](https://github.com/Yowkees/keyball/assets/164489400/fef6c150-4895-46d1-aec6-7a574e155c46)
![photo_2024-03-22_18-54-39](https://github.com/Yowkees/keyball/assets/164489400/d5d3a3fd-35b3-4162-9d5e-66c5429f276f)

After reading other posts from people with similar issues, I tried re-soldering the promicro and trrs, but the problem did not improve.

I asked an expert in my family, and he said there was something he couldn't understand from the circuit diagram. Below is what he wants to ask:

Looking at the circuit diagram, j13 TRS_JACK A B C D
B C is POWER and GND
A has no connection
And it is only connected to D, but it is only connected to DATA (RX)<-----------------> (RX)
 It can't communicate with this alone, but I don't understand this.

I would like to know how two-way communication works. It's probably one-way communication.

**Comment by eeJoe** [comment]
having exactly the same issue here - appreciate any help on this @Yowkees!

**Comment by koron** [comment]
Let me answer your questions first.

>  it is only connected to D, but it is only connected to DATA (RX)<-----------------> (RX)
It can't communicate with this alone

Keyball (QMK) uses half-duplex communication using one wire.
See also: https://docs.qmk.fm/#/serial_driver?id=bitbang
It only uses three wires in the TRRS cable.
So it also works with TRS cables (just three wires).

Next,  May I ask you some questions to help you solve the problem?

1. Please tell me the type and version of firmware you are using. 
2. If possible, use the latest 1.3.2 and test versions for subsequent steps.
3. Does it work if you connect a keyboard with a ball to the PC without connecting TRRS cable?
4. What if you did the same thing (step.3) with the keyboard without the ball?
5. If you connect the TRRS cable to the keyboard that has the ball, and connect it to the PC without connecting anything to the end of that TRRS cable, does it work?
6. What if you did the same thing (step.5) with the keyboard without the ball?
7. If you have both keyboards connected to the TRRS cable, and you connect the keyboard with the ball to the PC, please tell me exactly what will be displayed on both OLEDs.
8. Do the same thing as step 5, but replace the keyboard connected to the PC with one without a ball.

At  step 7 and 8, if it is functioning properly, various information should be displayed on the keyboard on the side connected to the PC, and the Keyball logo should be displayed on the keyboard on the other side.

**Comment by tmx976** [comment]
Same here.
I buyed a redy-to-use version form a vendor that didn't supply any basic instruction manual.
I connected left and right with TRSS cable, and the displays was as you state in the 7-8 step above.

After unplugged and plugged many times on some try, it has stop functioning if left and right are connected: to be more specific, each side works properly if the trss cable is not plugged in.
to answer to your question list:
1. how can i see that?
2.
3. yes (oled display many things)
4. yes (oled display many things)
5. no (as if it's without power, but it start working if I unplug the trss cable)
6. no (as if it's without power, but it start working if I unplug the trss cable)
7. no more
8. no more

Searching for this issue, I found that trss cable should'n be disconnected before usb-c cable.
I'm not sure, but I might have done it...

How can i recover from that?
Thank you.
